client/state/data-getters/index.js

```diff
@@ -282,16 +283,18 @@ export const requestTaxRate = ( countryCode, postalCode, httpOptions ) => {
 	} );
 };
 
-export const requestGutenbergBlockAvailability = siteSlug =>
-	requestHttpData(
+export const requestGutenbergBlockAvailability = siteSlug => {
+	const isBeta = isEnabled( 'jetpack/blocks/beta' ) ? '?beta=true' : '';
+	return requestHttpData(
 		`gutenberg-block-availability-${ siteSlug }`,
 		http(
 			{
-				path: `/sites/${ siteSlug }/gutenberg/available-extensions`,
+				path: `/sites/${ siteSlug }/gutenberg/available-extensions${ isBeta }`,
```

Comment:
Could we use `{ addQueryArgs } from 'lib/route'` instead of manually building the query string?